crates/flux-middle/src/fhir.rs

```diff
@@ -1069,6 +1081,21 @@ pub struct PathExpr<'fhir> {
     pub span: Span,
 }
 
+impl<'fhir> PathExpr<'fhir> {
+    pub fn new(
+        tcx: TyCtxt,
+        res: ExprRes,
+        segments: &'fhir [Ident],
+        fhir_id: FhirId,
+        span: Span,
+    ) -> Self {
+        if let Some(dst_span) = res.def_span(tcx) {
+            dbg::hyperlink!(tcx, span, dst_span);
+        }
```

Comment:
I don't like having the side effect here. All the places we create a `PathExpr` are in desugar, so we can have a method in the context.

Reply:
have moved this whole thing to `conv`, please take a look